### PR TITLE
CSP-778 fixes put in to stop null lat long or latitude of 0 being use…

### PR DESCRIPTION
**Pull request by MarkFCain**
…d in calculations

### CODE REVIEW

File: src/SFA.DAS.Roatp.Data/Repositories/ProviderDetailsReadRepository.cs [hunk 1]
```diff
@@ -83,8 +83,9 @@ private static FormattableString GetProviderForUkprnAndLarsCodeWithDistanceSql(i
                             p.Id as ProviderId,
                             CASE  WHEN ({lat} is null) THEN null
                                 WHEN ({lon} is null) THEN null
+                                WHEN (pa.Latitude is null) THEN null
                                 ELSE
-                                    geography::Point(isnull(pa.Latitude,0), isnull(pa.Longitude,0), 4326)
+                                    geography::Point(pa.Latitude, pa.Longitude, 4326)
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.Data/Repositories/ProviderDetailsReadRepository.cs:88]: Dont we need to check for 0 value here?

File: src/SFA.DAS.Roatp.Data/Repositories/ProviderDetailsReadRepository.cs [hunk 2]
```diff
@@ -108,8 +109,9 @@ private static FormattableString GetProvidersForLarsCodeWithDistanceSql(int lars
                                 ELSE pc.IsApprovedByRegulator End IsApprovedByRegulator,
                             CASE  WHEN ({lat} is null) THEN null
                                 WHEN ({lon} is null) THEN null
+                                WHEN (pa.Latitude is null) THEN null
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.Data/Repositories/ProviderDetailsReadRepository.cs:112]: same here?

**Review by DevenKShah — COMMENTED** [review]
Over all alignment is messed up in the queries, this needs to be fixed as it is unreadable

File: src/SFA.DAS.Roatp.Data/Repositories/ProviderDetailsReadRepository.cs (hunk 1 above)
Comment (MarkFCain) [reply src/SFA.DAS.Roatp.Data/Repositories/ProviderDetailsReadRepository.cs:88]: Well, I didn't find an occurence of0 in the data, so I wasn't too concerned, but to be extra safe, I will add that

File: src/SFA.DAS.Roatp.Data/Repositories/ProviderDetailsReadRepository.cs (hunk 2 above)
Comment (MarkFCain) [reply src/SFA.DAS.Roatp.Data/Repositories/ProviderDetailsReadRepository.cs:112]: see above

**Review by DevenKShah — APPROVED** [review]

**Pull request merged**